sycl/unittests/program_manager/passing_link_and_compile_options.cpp

```diff
@@ -0,0 +1,529 @@
+  std::string expected_compile_options1 = "-cl-single-precision-constant",
+              expected_compile_options1_1 = "",
+              expected_compile_options2 =
+                  "-cl-fp32-correctly-rounded-divide-sqrt",
+              expected_compile_options2_1 = "";
+  std::string expected_link_options = "";
+  static sycl::unittest::PiImage DevImage1 =
+      generateEAMTestKernel1Image<EAMTestKernel1>(expected_compile_options1,
+                                                  expected_link_options);
+  static sycl::unittest::PiImage DevImage2 =
+      generateEAMTestKernel1Image<EAMTestKernel2>(expected_compile_options2,
+                                                  expected_link_options);
+  static sycl::unittest::PiImageArray<1> DevImageArray_1{&DevImage1};
+  static sycl::unittest::PiImageArray<1> DevImageArray_2{&DevImage2};
+  auto KernelID_1 = sycl::get_kernel_id<EAMTestKernel1>();
+  auto KernelID_2 = sycl::get_kernel_id<EAMTestKernel2>();
+  sycl::queue Queue{Dev};
+  const sycl::context Ctx = Queue.get_context();
+  sycl::kernel_bundle KernelBundle1 =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(Ctx, {Dev},
+                                                         {KernelID_1});
+  auto BundleObj1 = sycl::compile(KernelBundle1);
+  sycl::link(BundleObj1);
+  EXPECT_EQ(expected_link_options, current_link_options);
+  EXPECT_EQ(expected_compile_options1, current_compile_options);
+  current_link_options.clear();
+  current_compile_options.clear();
+  sycl::kernel_bundle KernelBundle2 =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(Ctx, {Dev},
+                                                         {KernelID_2});
+  auto BundleObj2 = sycl::compile(KernelBundle2);
+  sycl::link(BundleObj2);
+  EXPECT_EQ(expected_compile_options2, current_compile_options);
+  EXPECT_EQ(expected_link_options, current_link_options);
+  // Check devices' compile & link options together
+  current_link_options.clear();
+  current_compile_options.clear();
+  static sycl::unittest::PiImageArray<2> DevImageArray[] = {&DevImage1,
+                                                            &DevImage2};
+  sycl::kernel_bundle KernelBundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(
+          Ctx, {Dev}, {KernelID_1, KernelID_2});
+  auto BundleObj = sycl::compile(KernelBundle);
+  sycl::link(BundleObj);
+  EXPECT_EQ("", current_link_options);
+  EXPECT_EQ(expected_compile_options1 + " " + expected_compile_options2,
+            current_compile_options);
+  // empty str + not empty str
+  DevImage1 = generateEAMTestKernel1Image<EAMTestKernel1>(
+      expected_compile_options1_1, expected_link_options);
+  current_link_options.clear();
+  current_compile_options.clear();
+  static sycl::unittest::PiImageArray<2> DevImageArray1[] = {&DevImage1,
+                                                             &DevImage2};
+
+  BundleObj = sycl::compile(KernelBundle);
+  sycl::link(BundleObj);
```

Comment:
Could you please tell why linking is done here? It seems the test doesn't check for link options?

Reply:
Yes, this test is checking only compile options. I'm rewriting tests now and I'm going to remove all unnecessary actions.